### PR TITLE
Home page frontend news

### PR DESCRIPTION
**Pull request by patofernandez**
## Summary
The functionality of the news was added
## Screenshot
image
## Trello
https://trello.com/c/oNQhz6i5
https://trello.com/c/Rf06OzTv

### CODE REVIEW

File: app/src/main/res/layout/item_news.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        android:layout_marginTop="16dp"
+        android:layout_marginEnd="16dp"
+        android:text="15m"
+        android:textSize="12sp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <ImageView
+        android:id="@+id/vFavorite"
+        android:layout_width="24dp"
```
Comment (DemianIac) [inline app/src/main/res/layout/item_news.xml]: This should be in the resource file

File: app/src/main/res/layout/item_news.xml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    xmlns:fresco="http://schemas.android.com/apk/res-auto">
+
+    <com.facebook.drawee.view.SimpleDraweeView
+        android:id="@+id/vAvatar"
+        android:layout_width="48dp"
```
Comment (DemianIac) [inline app/src/main/res/layout/item_news.xml]: This should be in the resource file

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt [hunk 3]
```diff
@@ -1,11 +1,40 @@
...
+    fun loadNews() {
+        retrofitServices.getService(NewsService::class.java).getNews().enqueue(object : Callback<List<News>> {
+            override fun onResponse(call: Call<List<News>>, response: Response<List<News>>) {
+                if (response.body()!!.isEmpty()) {
+                    toastFactory.show("There is no news to show")
+                } else {
+                    for (news in response.body()!!) {
+                        myNews.add(news)
+                    }
+                    for (i in 4..12) {
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt]: Always add a TODO to remove testing code.

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt [hunk 4]
```diff
@@ -1,17 +1,43 @@
...
+        presenter.loadNews()
     }
 
-    companion object {
-        fun newInstance() = NewsFragment()
+    override fun onNewsLoaded(myNews: ArrayList<News>) {
+        viewAdapter.setNews(myNews)
+        viewAdapter.notifyDataSetChanged()
+        if (vSwiperefresh.isRefreshing) {
+            toastFactory.show("The news was updated")
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt]: Try to get the texts from the resource. You can avoid this only if it's for debugging or something like that.

**Review by DemianIac — CHANGES_REQUESTED** [review]

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt (hunk 4 above)
Comment (patofernandez) [reply app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt]: done!

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt (hunk 3 above)
Comment (patofernandez) [reply app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt]: removed

File: app/src/main/res/layout/item_news.xml (hunk 1 above)
Comment (patofernandez) [reply app/src/main/res/layout/item_news.xml]: done

File: app/src/main/res/layout/item_news.xml (hunk 2 above)
Comment (patofernandez) [reply app/src/main/res/layout/item_news.xml]: done!

**Pull request closed without merging**